content/blog/2023/2023-08-24-fairness-reweighing-dataset.md

```diff
@@ -0,0 +1,41 @@
+
+### Reweighing:
+
+The Reweighing widget offers a solution to mitigate bias in datasets by assigning weights to individual instances. These weights are determined using the [Reweighing algorithm](https://link.springer.com/article/10.1007/s10115-011-0463-8). Essentially, the algorithm gives higher weights to underrepresented pairs of protected attributes and classes and reduces weights for overrepresented ones. This strategy encourages the model to prioritize learning from underrepresented groups while de-emphasizing overrepresented groups. In Orange, we can use the Reweighing widget in two ways:
+
+- **Reweighing a dataset:** The widget can be used to reweigh a dataset, which we can then use to train a model. Simply connect your dataset to the widget, and the output will be a dataset with appropriately weighted instances.
+
+- **Reweighing as a preprocessor:** The widget can function as a preprocessor for a specific model. By connecting the Reweighing widget directly to your chosen model, the dataset is automatically reweighed prior to the training phase.
+
+## Orange use case
+
+Now that we have grasped how the Reweighing widget functions and its applications, let us explore a real-world example of using it to assign weights to data. For this illustration, we will use the [Compas dataset](https://github.com/propublica/compas-analysis). Contrary to the example in the previous blog, we will not use the As Fairness widget to select fairness attributes. This is because datasets with a fairness tag come with default fairness attributes. Specifically, for the Compas dataset, "race" is identified as the protected attribute, with "Caucasian" set as the privileged, protected attribute value.
+
+{{< window-screenshot src="/blog_img/2023/2023-08-24-fairness-reweighing-dataset-use-case.png" >}}
+
+From the results, we can see that after reweighing the dataset, bias decreased substantially, nearing zero. Let us start by highlighting instances with lower or higher weights in the distribution widget to understand how the reweighing process assigned weights. Following this, we will analyze which combinations of protected attributes and classes were allocated these distinct weights.
+
+{{< window-screenshot src="/blog_img/2023/2023-08-24-fairness-reweighing-dataset-distributions.png" >}}
+
+{{< window-screenshot src="/blog_img/2023/2023-08-24-fairness-reweighing-dataset-box-plot.png" >}}
+
+The box plot widget reveals that lower weights were given to instances of unprivileged groups with unfavorable class values and privileged groups with favorable class values. This behavior aligns perfectly with the expected outcomes of the reweighing algorithm. The opposite is true for the higher weights. The results show that the reweighing algorithm successfully assigned weights to the instances in a way that will encourage the model to prioritize learning from underrepresented groups while de-emphasizing overrepresented groups.
```

Comment:
Where can we see weights in the boxplot? The only thing I see in the Boxplot is that every other race than Caucasian is predicted "Yes".

Reply:
Also comment on what this means in the context of the data set, refer to actual feature values. Your current description is too generic.